conans/client/profile_loader.py

```diff
@@ -263,7 +263,10 @@ def get_profile(profile_text, base_profile=None):
                                                          "options", "conf", "buildenv", "runenv",
                                                          "replace_requires", "replace_tool_requires",
                                                          "runner"])
-
+        if doc.duplicated_sections:
+            sections = ', '.join(sorted([f'[{s}]' for s in doc.duplicated_sections]))
+            ConanOutput().warning(f"Profile file has duplicated sections: {sections}. It could cause"
```

Comment:
Isnt this more of an actual error? I guess it would be breaking to raise :/

Reply:
I would prefer to see it as an error, as conan is ignoring some settings in the shadowed section.  That could be causing some extremely weird behaviour that would be hard to detect until much later.

Reply:
That's why I did not raise an error. I was afraid of breaking anything. IMO, a warning is enough to realize any unexpected mistake.

Reply:
@memsharded wdyt?

Reply:
I think warnings are often skipped and not seen at all.

As the behavior is basically destructive, adding a repeated section will destroy previous definition, this is a very bad thing, so I would raise an error if this condition is detected.